Dockerfile

```diff
@@ -0,0 +1,138 @@
+# ARG base_image=ubuntu:18.04
+ARG base_image=python:3.8
+FROM $base_image
+
+USER root
+
+# Version strings and default paths
+ENV SPARK_VERSION=2.4.5 \
+    HADOOP_VERSION=2.7 \
+    HADOOP_HOME=/usr/local/hdp \
+    SPARK_HOME=/usr/local/spark \
+    DBT_HOME=/usr/local/dbt-spark \
+    HOME=/home
+WORKDIR $HOME
+
+### Use spark/conf instead of hadoop/shared/conf:
+ENV HADOOP_CONF_DIR /usr/local/spark/conf
+
+# Env tuning
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONIOENCODING=utf-8 \
+    LANG=C.UTF-8
+
+# Install core libraries and dependencies
+RUN apt-get -y update && \
+    apt-get install -y \
+    apt-transport-https \
+    apt-utils \
+    ca-certificates \
+    ca-certificates-java \
+    iptables \
+    libcurl4-openssl-dev \
+    libsasl2-dev \
+    libssl-dev \
+    libxml2-dev \
+    lxc \
+    openssh-client \
+    net-tools \
+    wget && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install Java
+RUN apt-get -y update && \
+    apt-get install -y \
+    default-jre && \
+    rm -rf /var/lib/apt/lists/* && \
+    echo JAVA_HOME=$JAVA_HOME && \
+    echo "which java=`which java`" && \
+    java -version
+
+# RUN update-alternatives --config java
```

Comment:
why is that commented out ?

Reply:
Sorry - my mistake.  The call to `update-alternatives` isn't necessary. I'll remove. 

While debugging (before adding `which java`) I was trying to check if java was correctly configured (still not sure if JAVA_HOME is needed) but that should get flushed out once there's I've added tests,

Reply:
Removed.